src/EGIS.DynamoLib/ShapeFileReader.cs

```diff
@@ -62,10 +70,25 @@ public static Shape LoadFromFile(string shapeFilePath)
         /// <returns>List of PolyCurve</returns>
         public IEnumerable<Curve> GetShapeAtRecord(int index)
         {
-            return ShapeData[index].Select(pts => 
+            return ShapeData[index].Select(pts =>
             {
-                var points = pts.Select(p => Point.ByCoordinates(p.X, p.Y, p.Z));
-                var polygon = NurbsCurve.ByPoints(points);
+                var points = pts.Select(p => Point.ByCoordinates(p.X, p.Y, p.Z)).ToList();
+
+                //ctrl k d
+                {
+                    var pts0 = new List<Point>();
+                    for (int pi = 0; pi < points.Count - 1; ++pi)
+                    {
+                        pts0.Add(points[pi]);
+                        if (points[pi].IsAlmostEqualTo(points[pi + 1]))
+                        {
+                            ++pi;
+                        }
+                    }
+                    points = pts0;
```

Comment:
let's not overwrite points otherwise, the discarded points will not be disposed.